datalad/distribution/publish.py

```diff
@@ -15,108 +15,94 @@
+from .dataset import datasetmethod
+from .install import get_containing_subdataset
+
 
 lgr = logging.getLogger('datalad.distribution.publish')
 
 
 class Publish(Interface):
-    """Publish (e.g. to a web server) a dataset(s)
+    """Publish a dataset (e.g. to a web server)
+
+    This makes the current state of a dataset available to a sibling of the
+    dataset. That sibling needs to be known to the dataset before.
+    `create_publication_target` commands are meant to set up such a sibling, but
+    generally you can publish to any sibling added to a dataset.
+    Publishing may or may not include subdatasets.
+    By default file handles are published without their actual content,
+    but they can be published including the content, of course.
 
-    This is basic implementation for testing purposes
     """
+    # TODO: Figure out, how to tell about tracking branch/upstream
+    #      (and the respective remote)
+    #      - it is used, when no destination is given
+    #      - it is configured to be the given destination, if there was no
+    #        upstream set up before, so you can use just "datalad publish" next
+    #        time.
 
     _params_ = dict(
         dataset=Parameter(
             args=("-d", "--dataset"),
-            doc="""specify the dataset to perform the publish operation on.  If
-            no dataset is given, an attempt is made to identify the dataset
-            based on the current working directory and/or the `path` given""",
+            metavar='DATASET',
+            doc="""specify the dataset to publish. If no dataset is given, an
+            attempt is made to identify the dataset based on the current
+            working directory""",
             constraints=EnsureDataset() | EnsureNone()),
         dest=Parameter(
             args=("dest",),
-            doc="""url, local path, or sibling name identifying the publication
-            target""",
+            metavar='DEST',
+            doc="""sibling name identifying the publication target. If no
+            destination is given an attempt is made to identify the target based
+            on the dataset's configuration.""",
+            # TODO: See TODO at top of class!
             nargs="?",
             constraints=EnsureStr() | EnsureNone()),
-        path=Parameter(
-            args=("path",),
-            doc="path/name of the dataset component to publish",
-            nargs="*",
-            constraints=EnsureStr() | EnsureNone()),
         recursive=Parameter(
             args=("-r", "--recursive"),
             action="store_true",
-            doc="recursively publish all components of the dataset"),
+            doc="recursively publish all subdatasets of the dataset"),
         with_data=Parameter(
             args=("--with-data",),
-            doc="shell pattern",
+            metavar='PATH',
+            doc="path(s) to file handle(s) to publish including their actual "
+                "content",
```

Comment:
agree

2016-06-16 14:51 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/distribution/publish.py
> <https://github.com/datalad/datalad/pull/557#discussion_r67338056>:
>
> >          with_data=Parameter(
> >              args=("--with-data",),
> > -            doc="shell pattern",
> > +            metavar='PATH',
> > +            doc="path(s) to file handle(s) to publish including their actual "
> > +                "content",
>
> anyways -- majority of usecases would also be needing just passing args to annex
> copy to take advantage of its rich API and meta-data there. So, ideally,
> there should be right away the "--annex-copy-opts" to be channeled to annex
> copy.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/557/files/c118bd27c8e80aac8ffcb8bb42134f37d33c12e1#r67338056>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKAw_bbmNViZYPqMJzIl2XPiZkRu6Bxaks5qMUbYgaJpZM4I3KRD>
> .
>